src/V3Order.cpp

```diff
@@ -615,6 +615,43 @@ class MTaskVxIdLessThan final {
     }
 };
 
+//######################################################################
+// Gather variables written by a tree
+
+class OrderGatherWrittenVisitor final : public AstNVisitor {
+    std::unordered_set<AstVarScope*> m_writtenVariables;  // Variables written under tree
+
+    virtual void visit(AstVarRef* nodep) override {
+        if (nodep->access().isReadOnly()) return;
+        AstVarScope* const varScopep = nodep->varScopep();
+        if (varScopep->varp()->isFuncLocal()) return;
```

Comment:
Think ok but did you consider all the other var flags?

Reply:
I haven't considered all of them in great detail, but I believe this predicate is just a verilation time optimisation, because:

1. Variables only ever referenced under a CFunc will only be added to the order graph with a single edge from the input vertex to that OrderVarVertex (what this patch does), and will have no other out edges, so should not affect the actual ordering.

2. For any other variables (i.e.: those referenced outside the CFunc), we do want to the new edge

This means that conservatively we could add edges for all written variables in the CFunc, and still get the same ordering. Ignoring the locals just avoids adding these otherwise irrelevant vertices. (Will add comment to explain this).